### DISCUSSION TITLE
Template options "missingKey=error" forbids the use of "default" for a undefined value

### DISCUSSION THREAD

**Discussion by qwentb** [helmfile/helmfile#1825] [Questions]
In my gotmpl files, I tried to use syntax like 
```yaml
{{- $ingressClassName := default "nginx" .Values.global.ingress.defaultClass }}
```
but since template options  "missingKey" is set to "error" it obviously fails if ```.Values.global.ingress.defaultClass``` or one of is ancestors is not defined.
Is there a valid reason to enforce the  "missingKey=error" instead of the default "missingKey=Invalid" ?
Is there a workaround other than using ``` {{ if index .Values "global" ...``` for each level ?

**Comment by yxxhero** [comment]
you can use `get` template func. @QwentB

**Reply by qwentb** [reply]
Thanks for your answer @yxxhero but, I still have an issue with `{{ if .Value.global.some.missing.key }}` to be modified using `getOrNil`. 
I have quite a number of templates that usually work, and will need rewriting even if they have a valid go template syntax.

IMO it's not a good thing to forbid the "standard" use of very common functionnalities like `default` or `if ...`.
Would'nt it be possible to have an option to define the "missingkey" param value ?
